enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,286 @@
+
+    - UWM admins: manage the configuration of the UWM components (edit permissions on the openshift-user-workload-monitoring/user-workload-monitoring-config ConfigMap).
+    - Cluster admins: manage the configuration of the Platform monitoring components.
+
+In managed OpenShift clusters like OSD/ROSA, two separate CRDs are necessary because platform SREs manage the cluster's platform monitoring stack, while customers manage the user-defined monitoring stack. This separation ensures that each group maintains control over their specific monitoring configurations, reducing conflicts and enhancing system management.
+
+[More info](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/multi-tenant-alerting.md)
+
+
+- Replace configmaps with CRD:
+
+Current Config struct in CMO
+  
+```
+  type Config struct {
+	Images                               *Images `json:"-"`
+	RemoteWrite                          bool    `json:"-"`
+	CollectionProfilesFeatureGateEnabled bool    `json:"-"`
+
+	ClusterMonitoringConfiguration *ClusterMonitoringConfiguration `json:"-"`
+	UserWorkloadConfiguration      *UserWorkloadConfiguration      `json:"-"`
+}
+```
+
+We will strive to maintain the previous structure as much as possible while adapting it to OpenShift API standards. 
+
+Current ClusterMonitoringConfiguration struct:
+```
+type ClusterMonitoringConfiguration struct {
+	AlertmanagerMainConfig *AlertmanagerMainConfig `json:"alertmanagerMain,omitempty"`
+	UserWorkloadEnabled *bool `json:"enableUserWorkload,omitempty"`
+	HTTPConfig *HTTPConfig `json:"http,omitempty"`
+	K8sPrometheusAdapter *K8sPrometheusAdapter `json:"k8sPrometheusAdapter,omitempty"`
+	MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+	KubeStateMetricsConfig *KubeStateMetricsConfig `json:"kubeStateMetrics,omitempty"`
+	PrometheusK8sConfig *PrometheusK8sConfig `json:"prometheusK8s,omitempty"`
+	PrometheusOperatorConfig *PrometheusOperatorConfig `json:"prometheusOperator,omitempty"`
+	PrometheusOperatorAdmissionWebhookConfig *PrometheusOperatorAdmissionWebhookConfig `json:"prometheusOperatorAdmissionWebhook,omitempty"`
+	OpenShiftMetricsConfig *OpenShiftStateMetricsConfig `json:"openshiftStateMetrics,omitempty"`
+	TelemeterClientConfig *TelemeterClientConfig `json:"telemeterClient,omitempty"`
+	ThanosQuerierConfig *ThanosQuerierConfig `json:"thanosQuerier,omitempty"`
+	NodeExporterConfig NodeExporterConfig `json:"nodeExporter,omitempty"`
+	MonitoringPluginConfig *MonitoringPluginConfig `json:"monitoringPlugin,omitempty"`
+}
+```
+
+
+Each component within the ConfigMap will be migrated to the OpenShift API in separate PRs. This approach allows for a thorough review, improvement, and modification of each ConfigMap component to ensure it aligns with OpenShift API standards. As part of this process, types will be modified, outdated elements will be removed and names and configurations will be refined.
```

Comment:
Yeah that is the plan. Should this be mentioned here or are you looking for a more detailed plan?

Reply:
I would certainly mention it here, and, perhaps if possible, create a list of the ones you know are co-dependent and therefore should be considered together

Reply:
There are a lot components, not sure all the co-dependencies. The idea was to migrate one by one, and discuss in every PR the names, types, definitions and possible removes of types or refactors. After that we can think in handle the co-dependence. 

At the very beginning we tried to migrate all at the same time to see the similar fields and dependencies but its a huge task and no so useful.

Reply:
Right, our current assumption is that the configmap structure already separates the components pretty well. The only interdependency we have is that the `UserDefinedMonitoring` object will only have an effect if `ClusterMonitoringSpec.UserDefined != Disabled`, speaking in CRD terms. There is a similar interdependency for dedicated alertmanagers for user defined monitoring. It's probably worth calling out those two (soft) dependencies and our assumption that otherwise things are separated.